### PR TITLE
Add uint16 support for tf.decode_raw

### LINKED ISSUE #10124: [feature request] decode_raw for uint16 and other dtype
### System information
- NO
- NOT RELATED
- NOT RELATED
- NOT RELATED
- NOT RELATED
- NOT RELATED
- NOT RELATED
- NOT RELATED

### Describe the problem
Current decode_raw only supports `tf.half, tf.float32, tf.float64, tf.int32, tf.uint8, tf.int16, tf.int8, tf.int64`. It would be helpful if more types are available.

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the request raised in #10124 where uint16 support for tf.decode_raw is needed. tf.decode_raw already support half, float32, float64, int8, int16, int32, int64, uint8.

And uint16 was not supported yet.

This fix adds uint16 support for tf.decode_raw.

This fix fixes #10124.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by yongtang** [comment]
The previously Jenkins test failure `tfexample_decoder_test` was caused by the fact that `decode_raw` support `uint16` now.

The PR has been updated. Now Jenkins tests should pass.

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please

File: tensorflow/contrib/slim/python/slim/data/tfexample_decoder_test.py [hunk 1]
```diff
@@ -228,7 +228,10 @@ def testDecodeExampleWithJpegEncodingAt16BitCausesError(self):
     image_shape = (2, 3, 3)
     unused_image, serialized_example = self.GenerateImage(
         image_format='jpeg', image_shape=image_shape)
-    with self.assertRaises(TypeError):
+    # decode_raw support uint16 now so ValueError will be thrown instead.
```
Comment (rmlarsen) [inline tensorflow/contrib/slim/python/slim/data/tfexample_decoder_test.py]: Can you also add a unit test that tests that decode_raw can now successfully decode uint16?

**Review by rmlarsen — CHANGES_REQUESTED** [review]

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by yongtang** [comment]
@rmlarsen Thanks for the review. The PR has been updated with unit tests added. Please take a look.

**Review by rmlarsen — APPROVED** [review]

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Pull request merged**